src/session.ts

```diff
@@ -1,14 +1,10 @@
-import type {
-	Session,
-	SessionData,
-	SessionStorage,
-} from "@remix-run/server-runtime";
 import type { Context } from "hono";
+import type { Session, SessionData, SessionStorage } from "react-router";
 
 import { createMiddleware } from "hono/factory";
 
-const sessionStorageSymbol = Symbol();
-const sessionSymbol = Symbol();
+const sessionStorageSymbol = Symbol().toString();
+const sessionSymbol = Symbol().toString();
```

Comment:
This has an issue, `Symbol() === Symbol()` returns `false` but `Symbol().toString() === Symbol().toString()` returns `true`

If Hono doesn't allow symbols anymore, then these needs to be unique strings

Reply:
Tested on my end during react-router v7 migration using this branch and locally patching it back to `Symbol()`, and verified that `Symbol()` works just fine. My hono version is v4.6.4

Reply:
> This has an issue, `Symbol() === Symbol()` returns `false` but `Symbol().toString() === Symbol().toString()` returns `true`
> 
> If Hono doesn't allow symbols anymore, then these needs to be unique strings

Good catch!
To be honest, Hono accepts everything derived from `Env: { Variables }`, but since we don't pass a type to `createMiddleware`, it falls back to `{string: any}`.(https://github.com/honojs/hono/blob/bfc190f72573379565f6fab3f290e657f5778044/src/context.ts#L568).
So, I have added an `Env` type to pass to `createMiddleware` and I've checked that the user code doesn't complain about it.

_I've rolled back my commit with the string key because it can introduce conflicts if someone uses the same keys in their context._

@lhr0909 `Symbol` works in JS but the issue here is a pure type error. It would be kind if you could test this version and tell me if you have any issues with Context/Env 🙏

Reply:
@rphlmr I have tested and it works fine on my end. Thank you!